File: Location.php
<?php

function parseDate($input) {
    $d = DateTime::createFromFormat('d/m/Y', $input);
    return $d && $d->format('d/m/Y') === $input ? $d : false;
}

function addDays($date, $days) {
    $copy = clone $date;
    return $copy->modify("+$days days");
}

$items = [
    ["name" => "Voiture Kia Sportage 4", "reservations" => []],
    ["name" => "Appartement T2", "reservations" => []],
    ["name" => "Vélo de ville", "reservations" => []],
    ["name" => "Scooter SYM JET S", "reservations" => []],
    ["name" => "Set de vaisselle", "reservations" => []],
    ["name" => "Espace 800m²", "reservations" => []],
];

while (true) {
    echo "\n=== Réservation ===\n";
    foreach ($items as $i => $item) {
        echo ($i + 1) . ". " . $item["name"] . "\n";
    }

    echo "Ecrire ici le numéro qui contient votre choix : ";
    $choice = trim(fgets(STDIN));
    if (!is_numeric($choice) || $choice < 1 || $choice > count($items)) break;

    $index = $choice - 1;
    echo "Date jj/mm/yyyy: ";
    $dateInput = trim(fgets(STDIN));
    echo "Location pendant combien de jours ? : ";
    $daysInput = trim(fgets(STDIN));
    $startDate = parseDate($dateInput);
    $days = intval($daysInput);

    if (!$startDate || $days < 1) {
        echo "Entrée invalide.\n";
        continue;
    }

    $endDate = addDays($startDate, $days);
    $conflict = false;

    foreach ($items[$index]["reservations"] as $res) {
        if ($startDate < $res["end"] && $endDate > $res["start"]) {
            echo "Non disponible en ce moment. Prochain: " . $res["end"]->format("d/m/Y") . "\n";
            $conflict = true;
            break;
        }
    }

    if (!$conflict) {
        $items[$index]["reservations"][] = ["start" => $startDate, "end" => $endDate];
        echo "Réservé du " . $startDate->format("d/m/Y") . " au " . $endDate->format("d/m/Y") . "\n";
    }

    echo "SOuhaitez vous faire d'autre reservation ? (o/n): ";
    $rep = trim(fgets(STDIN));
    if (strtolower($rep) !== "o") break;
}

echo "Merci pour votre confiance, à bientôt !\n";

File: Location.cpp
#include <iostream>
#include <vector>
#include <string>
#include <ctime>
#include <sstream>
#include <iomanip>

bool parseDate(const std::string &s, std::tm &t)
{
    std::istringstream(s) >> std::get_time(&t, "%d/%m/%Y");
    return !std::istringstream(s).fail();
}

std::tm addDays(const std::tm &d, int days)
{
    auto tt = std::mktime(const_cast<std::tm *>(&d));
    tt += days * 86400;
    return *std::localtime(&tt);
}

struct Item
{
    std::string name;
    std::vector<std::pair<std::tm, std::tm>> res;
};

int main()
{
    std::vector<Item> items = {{"Voiture Kia Sportage 4"}, {"Appartement T2"}, {"Vélo de ville"}, {"Scooter SYM JET S"}, {"Set de vaisselle"}, {"Espace 800m²"}};
    char c;
    while (true)
    {
        std::cout << "=== Réservation ===\n";
        for (int i = 0; i < items.size(); i++)
            std::cout << i + 1 << ". " << items[i].name << "\n";
        std::cout << "Écrivez ici le numéro du type de location que vous souhaitez : ";
        int i;
        std::cin >> i;
        if (!std::cin || i < 1 || i > items.size())
            break;
        auto &it = items[i - 1];
        std::tm d = {};
        std::cout << "Date jj/mm/yyyy: ";
        std::cin >> std::get_time(&d, "%d/%m/%Y");
        int days;
        std::cout << "Pour combien de jours ? : ";
        std::cin >> days;
        auto start = d, end = addDays(d, days);
        bool ok = true;
        time_t s0 = std::mktime(&start), e0 = std::mktime(&end);
        for (auto &p : it.res)
        {
            time_t s1 = std::mktime(&p.first), e1 = std::mktime(&p.second);
            if (s0 < e1 && e0 > s1)
            {
                std::cout << "Non disponible, Prochain: " << std::put_time(&p.second, "%d/%m/%Y") << "\n";
                ok = false;
                break;
            }
        }
        if (ok)
        {
            it.res.emplace_back(start, end);
            std::cout << "Réservation du " << std::put_time(&start, "%d/%m/%Y") << " au "
                      << std::put_time(&end, "%d/%m/%Y") << "\n";
        }
        std::cout << "Souhaitez-vous faire d'autre réservation ? (o/n): ";
        std::cin >> c;
        if (c != 'o')
            break;
    }
}
